snuba/api.py

```diff
@@ -162,16 +162,80 @@ def query(validated_body=None, timer=None):
     )
 
 
+def format_query(dataset, body, table, prewhere_conditions, final) -> str:
+    """Generate a SQL string from the parameters."""
+    aggregate_exprs = [util.column_expr(dataset, col, body, alias, agg) for (agg, col, alias) in body['aggregations']]
+    groupby = util.to_list(body['groupby'])
+    group_exprs = [util.column_expr(dataset, gb, body) for gb in groupby]
+    selected_cols = [util.column_expr(dataset, util.tuplify(colname), body) for colname in body.get('selected_columns', [])]
+    select_clause = u'SELECT {}'.format(', '.join(group_exprs + aggregate_exprs + selected_cols))
+
+    from_clause = u'FROM {}'.format(table)
+    if final:
+        from_clause = u'{} FINAL'.format(from_clause)
+    if 'sample' in body:
+        from_clause = u'{} SAMPLE {}'.format(from_clause, body['sample'])
+
+    join_clause = ''
+    if 'arrayjoin' in body:
+        join_clause = u'ARRAY JOIN {}'.format(body['arrayjoin'])
```

Comment:
do we have to "unicodify" this one ?

Reply:
I'm not sure what this question means. I don't think we need to do any specific encoding here or anything. This will be a unicode string. The `u` prefix is actually unnecessary for both of these strings due to Python 3. Behaviorally, this is really just the same as before, with the exception that this removes some unnecessary list operations that weren't valid per the JSON schema.